var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -131,61 +141,78 @@ def headers(self):
             '-DLAPACKE:BOOL=%s' % (
-                'ON' if '+lapacke' in spec else 'OFF'),
+                'ON' if '+lapacke' in self.spec else 'OFF'),
             '-DLAPACKE_WITH_TMG:BOOL=%s' % (
-                'ON' if '+lapacke' in spec else 'OFF')]
-        if spec.satisfies('@3.6.0:'):
-            cmake_args.extend(['-DCBLAS=ON'])  # always build CBLAS
+                'ON' if '+lapacke' in self.spec else 'OFF')]
+
+        if self.spec.satisfies('@3.6.0:'):
+            args.extend(['-DCBLAS=ON'])  # always build CBLAS
 
-        if self.compiler.name == 'intel':
+        if self.spec.satisfies('%intel'):
             # Intel compiler finds serious syntax issues when trying to
             # build CBLAS and LapackE
-            cmake_args.extend(['-DCBLAS=OFF'])
-            cmake_args.extend(['-DLAPACKE:BOOL=OFF'])
+            args.extend(['-DCBLAS=OFF'])
+            args.extend(['-DLAPACKE:BOOL=OFF'])
 
-        if self.compiler.name == 'xl' or self.compiler.name == 'xl_r':
+        if self.spec.satisfies('%xl') or self.spec.satisfies('%xl_r'):
             # use F77 compiler if IBM XL
-            cmake_args.extend([
+            args.extend([
                 '-DCMAKE_Fortran_COMPILER=%s' % self.compiler.f77,
                 '-DCMAKE_Fortran_FLAGS=%s' % (
                     ' '.join(self.spec.compiler_flags['fflags'])),
             ])
 
         # deprecated routines are commonly needed by, for example, suitesparse
         # Note that OpenBLAS spack is built with deprecated routines
-        cmake_args.extend(['-DBUILD_DEPRECATED:BOOL=ON'])
+        args.extend(['-DBUILD_DEPRECATED:BOOL=ON'])
 
-        if '+external-blas' in spec:
-            cmake_args.extend([
+        if '+external-blas' in self.spec:
+            args.extend([
                 '-DUSE_OPTIMIZED_BLAS:BOOL=ON',
-                '-DBLAS_LIBRARIES:PATH=%s' % spec['blas'].libs.joined(';')
+                '-DBLAS_LIBRARIES:PATH=%s' % self.spec['blas'].libs.joined(';')
             ])
 
-        if spec.satisfies('+xblas'):
-            xblas_include_dir = spec['netlib-xblas'].prefix.include
-            xblas_library = spec['netlib-xblas'].libs.joined(';')
-            cmake_args.extend([
+        if self.spec.satisfies('+xblas'):
+            xblas_include_dir = self.spec['netlib-xblas'].prefix.include
+            xblas_library = self.spec['netlib-xblas'].libs.joined(';')
+            args.extend([
                 '-DXBLAS_INCLUDE_DIR={0}'.format(xblas_include_dir),
                 '-DXBLAS_LIBRARY={0}'.format(xblas_library)])
 
-        cmake_args.extend(std_cmake_args)
+        args.append('-DBUILD_TESTING:BOOL=%s' %
+                    ('ON' if self.run_tests else 'OFF'))
+
+        return args
+
+    # Build, install, and check both static and shared versions of the
+    # libraries when +shared
+    @when('+shared')
+    def cmake(self, spec, prefix):
+        for self._building_shared in range(2):
```

Comment:
Can you replace `range(2)` with `(True, False)`?

From [The Zen of Python](https://www.python.org/dev/peps/pep-0020/):

> Explicit is better than implicit.

Reply:
As you wish. I wouldn't say that `False=0` and 'True=1` is really implicit though. Especially in the context of creating unnecessary lists (at least in Python 3).